src/engine/client/textrender.h

```diff
@@ -0,0 +1,232 @@
+public:
+	CAtlas() { m_LastFrameAccess = 0; m_Access = 0; }
+	void Init(int Index, int X, int Y, int Width, int Height);
+
+	ivec2 Add(int Width, int Height);
+
+	int GetWidth() { return m_Width; }
+	int GetHeight() { return m_Height; }
+	int GetOffsetX() { return m_Offset.x; }
+	int GetOffsetY() { return m_Offset.y; }
+
+	int GetPageID() { return m_ID; }
+	void Touch() { m_Access++; }
+	int GetAccess() { return m_LastFrameAccess; }
+	void Update() { m_LastFrameAccess = m_Access; m_Access = 0; }
+};
+
+class CGlyphMap
+{
+	IGraphics *m_pGraphics;
+	FT_Stroker m_FtStroker;
+	IGraphics::CTextureHandle m_aTextures[2];
+	CAtlas m_aAtlasPages[PAGE_COUNT*PAGE_COUNT];
+	int m_ActiveAtlasIndex;
+	sorted_array<CGlyphIndex> m_Glyphs;
+
+	int m_NumTotalPages;
+
+	FT_Face m_DefaultFace;
+	FT_Face m_VariantFace;
+	FT_Face m_aFallbackFaces[MAX_FACES];
+	int m_NumFallbackFaces;
+
+	FT_Face m_aFtFaces[MAX_FACES];
+	int m_NumFtFaces;
+
+	int AdjustOutlineThicknessToFontSize(int OutlineThickness, int FontSize);
+
+	void InitTexture(int Width, int Height);
+	int FitGlyph(int Width, int Height, ivec2 *Position);
+	void UploadGlyph(int TextureIndex, int PosX, int PosY, int Width, int Height, const unsigned char *pData);
+	bool SetFaceByName(FT_Face *pFace, const char *pFamilyName);
+	int GetCharGlyph(int Chr, FT_Face *pFace);
+public:
+	CGlyphMap(IGraphics *pGraphics, FT_Library FtLibrary);
+	~CGlyphMap();
+
+	IGraphics::CTextureHandle GetTexture(int Index) { return m_aTextures[Index]; }
+	FT_Face GetDefaultFace() { return m_DefaultFace; };
+	int AddFace(FT_Face Face);
+	void SetDefaultFaceByName(const char *pFamilyName);
+	void AddFallbackFaceByName(const char *pFamilyName);
+	void SetVariantFaceByName(const char *pFamilyName);
+	
+	bool RenderGlyph(CGlyph *pGlyph, bool Render);
+	CGlyph *GetGlyph(int Chr, int FontSizeIndex, bool Render);
+	int GetFontSizeIndex(int PixelSize);
+	vec2 Kerning(CGlyph *pLeft, CGlyph *pRight, int PixelSize);
+
+	int NumTotalPages() { return m_NumTotalPages; }
+	void PagesAccessReset();
+};
+
+struct CFontLanguageVariant
+{
+	char m_aLanguageFile[128];
+	char m_aFamilyName[128];
+};
+
+struct CWordWidthHint
+{
+	float m_EffectiveAdvanceX;
+	int m_CharCount;
+	int m_GlyphCount;
+	bool m_EndOfWord;
+	bool m_EndsWithNewline;
+	bool m_IsBroken;
+};
+
+class CTextRender : public IEngineTextRender
+{
+	IGraphics *m_pGraphics;
+	IGraphics *Graphics() { return m_pGraphics; }
+
+	float m_TextR;
+	float m_TextG;
+	float m_TextB;
+	float m_TextA;
+
+	float m_TextSecondaryR;
+	float m_TextSecondaryG;
+	float m_TextSecondaryB;
+	float m_TextSecondaryA;
+
+	CGlyphMap *m_pGlyphMap;
+	void *m_apFontData[MAX_FACES];
+
+	// support regional variant fonts
+	int m_NumVariants;
+	int m_CurrentVariant;
+	CFontLanguageVariant *m_paVariants;
+
+	FT_Library m_FTLibrary;
+
+	int LoadFontCollection(const void *pFilename, const void *pBuf, long FileSize);
+
+	bool isWestern(int Chr)
+	{
+		if(Chr >= 0x0020 && Chr <= 0x218F)
```

Comment:
Just `return Chr >= 0x0020 && Chr <= 0x218F`

The function could be declared `const` for good measure.